### PR TITLE
[GH-21586] - Removed deprecated windows build function in Makefile

### PR DESCRIPTION
**Pull request by mattlam88**
#### Summary
The PR includes the removal of the deprecated build-windows command in the Makefile. Additionally, there is an update to the documentation that windows is no longer supported for make commands.

Please let me know if there is any additional documentation spots that may be a good candidate to add this change, as this is my first OSS PR!

#### Ticket Link
Fixes the following Github issue: https://github.com/mattermost/mattermost-server/issues/21586

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ make package
 
 We use a combination of `Makefile` and [Goreleser](https://goreleaser.com/) to automate our release process.
 
+NOTE: Makefile no longer supports building the binary file using Windows on AMD64
+
```
Comment (agarciamontoro) [inline docs/developing.md:32]: This is true, but it's not the only problem. The binary cannot be built for Windows at all now (we use some unix-specific stuff), not only because it's not in Makefile. We can reword this to be more generic. Something like "The load-test tool does not support Windows". And maybe putting somewhere else, not under the `Releasing` section. Even at the beginning of this file. Thoughts, @mattlam88?

File: Makefile [hunk 2]
```diff
@@ -35,11 +35,6 @@ build-osx: ## Build the binary (only for OSX on AMD64).
 	env GOOS=darwin GOARCH=amd64 $(GO) build -o $(AGENT) $(AGENT_ARGS)
 	env GOOS=darwin GOARCH=amd64 $(GO) build -o $(API_SERVER) $(API_SERVER_ARGS)
 
-build-windows: ## Build the binary (only for Windows on AMD64).
-	@echo Build Windows amd64
-	env GOOS=windows GOARCH=amd64 $(GO) build -o $(AGENT) $(AGENT_ARGS)
-	env GOOS=windows GOARCH=amd64 $(GO) build -o $(API_SERVER) $(API_SERVER_ARGS)
-
```
Comment (agarciamontoro) [inline Makefile:42]: :tada:

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thank you for your work here, @mattlam88 :tada: I'm glad your first contribution to OSS is for this repo! I left a comment below, thanks!

File: docs/developing.md (hunk 1 above)
Comment (mattlam88) [reply docs/developing.md:32]: That makes sense I'll update the PR to include that at the top

File: docs/developing.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
 ## Developer's workflow
+### Notes
+The load-test tool does not support Windows
```
Comment (agarciamontoro) [inline docs/developing.md:3]: Just a super small nit:
```suggestion
The load-test tool does not support Windows.
```

**Review by agarciamontoro — APPROVED** [review]
Cool! Thanks, @mattlam88 :tada:
